src/python/pants/source/filespec.py

```diff
@@ -0,0 +1,59 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import re
+
+
+def glob_to_regex(pattern):
+  """Given a glob pattern, return an equivalent regex expression.
+  :param string glob: The glob pattern. "**" matches 0 or more dirs recursively.
```

Comment:
glob syntax used in file matching is not equivalent to gitignore syntax. For example, "src" in gitignore will match everything inside "src", but in glob syntax, it should only match "src".